### PR TITLE
3.0

### PR DESCRIPTION
**Pull request by marianozicavo**
- Use `yup` for validation
- Change `createTypes` API https://github.com/Wolox/redux-recompose/pull/66
- Change `completeState` API https://github.com/Wolox/redux-recompose/pull/67
- Update dependencies
- `injections` must always be an array
- Run linter
- Set up github actions instead of previous wolox ci

### CODE REVIEW

File: src/completers/completeReducer/README.md [hunk 1]
```diff
@@ -1,56 +1,51 @@
...
+  [actions.POLLING_ACTION_FAILURE]: onFailure(),
+  [actions.POLLING_ACTION_RETRY]: onRetry(),
+  [actions.POLLING_ACTION_CANCEL]: onCancel(),
+  [actions.IGNORED_ACTION]: (state, action) => ({ ...state, someTarget: action.payload }),
+  [actions.CUSTOM_ACTION_COMPLETED]: (state, action) => ({ ...state, someOtherTarget: action.payload }),
+  [actions.PRIMARY_ACTION_FAILURE]: (state, action) => ({ ...state, someOtherTarget: action.payload })
   }
-}
-```
+*/
```
Comment (sfernandez11) [inline src/completers/completeReducer/README.md:48]: Is it ok all these commented code?

File: src/completers/completeTypes/README.md [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
 
-This completer helps to extends a group of actions including their `SUCCESS` and `FAILURE` cases. It also receives a second parameter that describes which ones are not extended.
+This completer can extend the list of possible action types, helping to reduce its code size or generate them programatically. `completeTypes` receives an object with:
 
-Examples:
+* `primaryActions`: The completer generates the `_SUCCESS` and `_FAILURE` actions.
+* `ignoredActions`: The completer doesn't generate any extra actions for these.
+* `pollingActions`: The completer generates the `_SUCCESS`, `_FAILURE`, `_RETRY` and `_CANCEL` actions.
+* `customCompleters`: You can specify what types to generate.
+
```
Comment (sfernandez11) [inline src/completers/completeTypes/README.md:9]: What's happened with `modalActions`?

File: src/completers/completeTypes/README.md [hunk 3]
```diff
@@ -1,20 +1,49 @@
...
+  "SECOND_POLLING_ACTION_RETRY",
+  "SECOND_POLLING_ACTION_CANCEL",
+  "FIRST_CUSTOM_ACTION",
+  "UPGRADED_FIRST_CUSTOM_ACTION",
+  "SECOND_CUSTOM_ACTION",
+  "UPGRADED_SECOND_CUSTOM_ACTION",
+  "NEW_THIRD_CUSTOM_ACTION",
+  "FIRST_IGNORED_ACTION",
+  "SECOND_IGNORED_ACTION"
+*/
```
Comment (sfernandez11) [inline src/completers/completeTypes/README.md:46]: Same with these examples, are necessary?

File: src/completers/completeTypes/index.js [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
+    completer: yup.mixed().test(value => typeof value === 'function').typeError('completer should be a function'),
+    actions: yup.array().of(yup.string().typeError('actions should be an array of strings')).typeError('actions should be an array')
+  }))
+}).typeError('reducerDescription should be an object');
 
-  return [...completedTypes, ...ignoredActions];
+function completeTypes(params) {
+  schema.validateSync(params);
+  const {
+    primaryActions = [], ignoredActions = [], customCompleters = [], pollingActions = []
```
Comment (sfernandez11) [inline src/completers/completeTypes/index.js:18]: Same here, in the `completeReducer` README you talk about `modalActions`. What happened?

File: src/effects/onRetry/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+Example:
+
+```js
+const selector =
+  (action, state) => action.payload.customError || state.defaultError;
+
+const reducerDescription = {
+  'SUCCESS': onSuccess(),
+  'SUCCESS_CUSTOM': onSuccess(selector)
```
Comment (sfernandez11) [inline src/effects/onRetry/README.md:23]: Is `SUCCESS` ok here? Why not `onRetry`?

**Review by sfernandez11 — COMMENTED** [review]
Amazing work Marian! Some comments...

File: recipes/FETCHING.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Making HTTP requests by dispatching a redux action
+`redux-recompose` enables you to make HTTP requests just by dispatching a redux action. Not only that, it also updates your state so you can know when you are waiting for the response and wether an error ocurred.
+
+You can check out [this great medium](https://medium.com/wolox/932e746b0198) post by [mvbattan](https://github.com/mvbattan) in order to understand how it works.
+
+You can try out an invisible reducer in a [demo](https://codesandbox.io/s/primary-actions-example-hl0b4) as well. We recommend using you own browser's console instead of `codesandbox.ios`'s.
```
[the remark this answers is not in the document]
Comment (marianozicavo) [reply recipes/FETCHING.md:6]: Yes it's ok.
If you look the code of the linked sandbox, it's not an invisible reducer. It's just that the example is almost the same.

File: src/completers/completeReducer/README.md (hunk 1 above)
Comment (marianozicavo) [reply src/completers/completeReducer/README.md:48]: It's there to show the content of `completedReducer`. I'll add an extra comment to make it clearer

File: src/completers/completeTypes/README.md (hunk 2 above)
Comment (marianozicavo) [reply src/completers/completeTypes/README.md:9]: I'll just delete them, actually. They are not documented and never really used

Comment (marianozicavo) [reply src/completers/completeTypes/README.md:9]: Same with `onSuscribe` and `onUnsuscribe` I think

File: src/effects/onRetry/README.md (hunk 5 above)
Comment (marianozicavo) [reply src/effects/onRetry/README.md:23]: You're right!

**Review by sfernandez11 — APPROVED** [review]
🥇 👏

**Pull request merged**